docs/source/developers/release.rst

```diff
@@ -36,8 +39,8 @@ Preparing for the release
 =========================
 
 Before creating a source release, the Release Manager must ensure that any
-resolved JIRAs have the appropriate Fix Version set so that the changelog is
-generated properly.
+resolved GitHub issues and any MINOR pull requests have the appropriate
+milestone set so that the changelog is generated properly.
```

Comment:
I grep'ed through the file and it seems we mention nowhere that minor prs have to be manually cherry picked as archery requires an issue (right? a quick glance at the code seems to confirm that).

Reply:
MINOR issues should not be cherry-picked for a release. If we have done the feature freeze and an issue is required for the release, the issue deserves to be non-MINOR.
We should not encourage the use of minor for issues that have to be cherry-picked.

Reply:
Good catches, thanks.

I think we could update the text then to say,

> Before creating a source release, the Release Manager must ensure that any
resolved GitHub issues have the appropriate milestone set so that the changelog
is generated properly.

Have there been cases where this is needed? Could there be a way to automate a check of some sort here?

Reply:
Now that I just made this comment, I think that's what the curate command currently does.

Reply:
> We should not encourage the use of minor for issues that have to be cherry-picked.

Agree on that but I think this might still be an issue for minor releases where having MINOR PRs should be ok (e.g. dependabot, ci changes etc.)? Or am I missing something @raulcd ?

Reply:
The problem is that manually going over those issues is error prone and it'll depend on the release manager. I am ok with dependabot but ci changes should not be minor, from our minor definition it should only be minor typos.
We could always add something in the lines:
> Pull requests without a corresponding GitHub issue (MINOR) won't be detected by our cherry-pick script. Some manual commits like dependabot won't have a corresponding issue. If necessary those commits should be cherry-picked manually but it is encouraged to create an issue if a change might require to be added to a release in order to avoid manual intervention for cherry-picking.

Reply:
Sounds good, though I think we should discuss the obligatory issue requirement in the future :) (I guess the main issue outside of the policy discussion on the ML is the adaptation of archery to handle it?)

Reply:
I made an attempt at adding a note like the quoted text above in https://github.com/apache/arrow/pull/45141/commits/d8cff77223db4a5881e6fb6301311aed3e6fcbde. Can either/both of you have a look and check if I understood correctly?

Reply:
Looks good to me. Thanks @amoeba

Reply:
Great. I'll merge this and put it into 19.0.0.